src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +691,433 @@ protected function limit_list( $key, $list ) {
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function set_attendee_args( $attendee_data, $ticket = null ) {
+		$args = [
+			'attendee_id'       => null,
+			'title'             => null,
+			'full_name'         => null,
+			'email'             => null,
+			'ticket_id'         => $ticket ? $ticket->ID : null,
+			'post_id'           => $ticket ? $ticket->get_event_id() : null,
+			'order_id'          => null,
+			'order_attendee_id' => null,
+			'user_id'           => null,
+			'attendee_status'   => null,
+			'price_paid'        => null,
+			'optout'            => null,
+		];
+
+		$args = array_merge( $args, $attendee_data );
+
+		$attendee_id = null;
+
+		// Unset the attendee ID if found.
+		if ( isset( $args['attendee_id'] ) ) {
+			$attendee_id = $args['attendee_id'];
+
+			unset( $args['attendee_id'] );
+		}
+
+		// Do some extra set up if creating an attendee.
+		if ( null === $attendee_id ) {
+			// Default attendees to opted-out.
+			if ( null === $args['optout'] ) {
+				$args['optout'] = 1;
+			}
+
+			// If the title is empty, set the title from the full name.
+			if ( empty( $args['title'] ) && $args['full_name'] ) {
+				$args['title'] = $args['full_name'];
+
+				// Maybe add the Order ID.
+				if ( $args['order_id'] ) {
+					$args['title'] = $args['order_id'] . ' | ' . $args['title'];
+				}
+
+				// Maybe add the Order Attendee ID.
+				if ( null !== $args['order_attendee_id'] ) {
+					$args['title'] .= ' | ' . $args['order_attendee_id'];
+				}
+			}
+
+			// Maybe handle setting the User ID based on information we already have.
+			if ( empty( $args['user_id'] ) && ! empty( $args['email'] ) ) {
+				/**
+				 * Allow filtering whether to enable user lookups by Attendee Email.
+				 *
+				 * @since TBD
+				 *
+				 * @param boolean $lookup_user_from_email Whether to lookup the User using the Attendee Email if User ID is not set.
+				 * @param array   $args                   The arguments being set for this attendee.
+				 */
+				$lookup_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_lookup_user_from_email', false, $args );
+
+				if ( $lookup_user_from_email ) {
+					// Check if user exists.
+					$user = get_user_by( 'email', $args['email'] );
+
+					if ( $user ) {
+						$args['user_id'] = $user->ID;
+					}
+				}
+
+				// Maybe create the user based on information we already have.
+				if ( empty( $args['user_id'] ) ) {
+					/**
+					 * Allow filtering whether to enable creating users using the Attendee Email.
+					 *
+					 * @since TBD
+					 *
+					 * @param boolean $create_user_from_email Whether to create the User using the Attendee Email if User ID is not set.
+					 * @param array   $args                   The arguments being set for this attendee.
+					 */
+					$create_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_create_user_from_email', false, $args );
```

Comment:
:no_entry_sign: Short form type keywords must be used. Found: (boolean)

<sub>hash: dbe16c5d7758aa9c66b39269a210eede</sub>